router/pkg/config/config.go

```diff
@@ -249,9 +249,20 @@ type CDNConfiguration struct {
 	CacheSize BytesString `yaml:"cache_size,omitempty" envconfig:"CDN_CACHE_SIZE" default:"100MB"`
 }
 
+type NatsAuthentication struct {
```

Comment:
I'd split it in `TokenBasedAuthenthication` and `UsernamePasswordAuthenthication` and inline both in `Authentication`

Reply:
How do you feel about `CredentialsBasedAuthentication` rather than `UsernamePasswordAuthentication`?